### PR TITLE
Allow the use of DIDs in the Cache API

### PR DESCRIPTION
**Pull request by lsenta**
- [x] Implement the code required to find stores from DIDs,
- [x] Add did-related parameters to `getProfile`, `getProfiles`, `list-spaces` and `space`,
- [x] Factor-out code heavily reused,
- [x] Test, test, test,
- [ ] stagings & production: check that we connect to infura nodes (where did manifests are stored).

### CODE REVIEW

**Comment by lsenta** [comment]
@oed @zachferland 
Regarding the last point: "stagings & production: check that we connect to infura nodes (where did manifests are stored)"

I tried with the default config: I can access my did files. I couldn't find a way to connect explicitly to the infura nodes (the docs describes going through their http API, not connecting an IPFS nodes to their network).

**Comment by oed** [comment]
@lsenta we can get the ID of the infura node by running `curl https://ipfs.infura.io:5001/api/v0/id`. Then I think we can just connect to one of the addresses listed there.

**Comment by lsenta** [comment]
Hou that's a good idea! But these are internal IPs

```json
  "Addresses": [
    "/ip4/127.0.0.1/tcp/4001/ipfs/QmXsiy5my3UummnhXeGn5vf8yUyBG8cjvfUeUY1JgydwLz",
    "/ip4/10.0.25.140/tcp/4001/ipfs/QmXsiy5my3UummnhXeGn5vf8yUyBG8cjvfUeUY1JgydwLz",
    "/ip4/172.17.0.1/tcp/4001/ipfs/QmXsiy5my3UummnhXeGn5vf8yUyBG8cjvfUeUY1JgydwLz",
    "/ip4/3.91.3.164/tcp/4001/ipfs/QmXsiy5my3UummnhXeGn5vf8yUyBG8cjvfUeUY1JgydwLz"
  ],
```

I tried the swarm API https://ipfs.infura.io:5001/api/v0/swarm/peers it throws a 403, I've opened a ticket to infura to ask for more info.

**Comment by oed** [comment]
I'll try to just ask the directly :)

File: package.json [hunk 1]
```diff
@@ -30,7 +30,10 @@
     "express": "^4.16.4",
     "get-folder-size": "^2.0.0",
     "ipfs": "^0.33.1",
-    "orbit-db": "^0.19.9",
+    "js-sha256": "^0.9.0",
+    "keccak": "^2.0.0",
```
Comment (oed) [inline package.json]: Looks like this is not used.

File: package.json [hunk 2]
```diff
@@ -30,7 +30,10 @@
...
+    "multihashes": "^0.4.14",
+    "orbit-db": "git://github.com/orbitdb/orbit-db.git#941c078",
```
Comment (oed) [inline package.json]: I think we are going with the version specified in this PR: https://github.com/3box/3box-pinning-server/pull/107

File: src/cacheService.js [hunk 3]
```diff
@@ -21,91 +23,133 @@ class CacheService {
...
-    try {
-      getRes = await axios.get(request)
-    } catch (e) {
-      res.status(404).send({ status: 'error', message: 'Address link not found, address does not have a 3Box or is malformed' })
-      return
-    }
-    const rootStoreAddress = getRes.data.data.rootStoreAddress
-    const cacheSpaces = await this.cache.read(`space-list_${rootStoreAddress}`)
-    let spaces
+    const { address, did } = req.query
```
Comment (oed) [inline src/cacheService.js]: Neat :)

File: package.json (hunk 1 above)
Comment (lsenta) [reply package.json]: Good catch, thanks!

File: package.json (hunk 2 above)
Comment (lsenta) [reply package.json]: Done!

**Pull request merged**